libcxx/utils/libcxx/test/format.py

```diff
@@ -132,13 +154,62 @@ def parseScript(test, preamble):
     script += scriptInTest
 
     # Add compile flags specified with ADDITIONAL_COMPILE_FLAGS.
+    # Modules need to be build with the same compilation flags as the
+    # test. So add these flags before adding the modules.
     substitutions = [
```

Comment:
Suggestion: Use `substitutions = dsl._appendToSubstitution(substitutions, "%{compile_flags}", " ".join(additionalCompileFlags))`

Same for the modules compile flags.